group/assignment-6/rearranging_cars.h

```diff
@@ -0,0 +1,59 @@
+#ifndef REARRANGING_CARS_H_
+#define REARRANGING_CARS_H_
+
+#include <string>
+#include <vector>
+#include <set>
+#include <utility>
+
+struct Car {
+  std::string id;
+  int slotId;
+  Car(std::string id);
+};
+
+struct Move {
+  std::string carId;
+  int fromSlotId;
+  int toSlotId;
+};
+
+using ParkingLotState = std::vector<Car*>;
+using Solution = std::pair<bool, std::vector<Move> >;
+
+
+class RearrangingCars {
+  public:
+    RearrangingCars(ParkingLotState startState, ParkingLotState endState);
+
+    void printSolution();// will remove it after @Huihui writes the testing files
+
+    void findSolution();// will remove it after @Huihui writes the testing files
+
+    Solution rearrange();
+
+  private:
+    int emptySlotPos = -1;

```

Comment:
Put that -1 value in a constant variable `kInvalidSlotId` and use it in the code when needed.

```suggestion
static const int kInvalidSlotId  = -1;
```

Reply:
Done.